File: Tupac.java
package tests;

import java.util.Scanner;

public class Tupac {

	public static void main(String[] args) {
		
		Scanner sc = new Scanner(System.in);
		System.out.println("Enter index");
		
		int lim = sc.nextInt() , dig , itw , rem , len;
		String num = "1", newt = "";
		
		for(int i = 1 ; i <= lim ; i++){
			
			newt = "";
			num = "0" + num;
			len = num.length() -1 ;
			rem = 0;
			
			while(len >= 0){
				dig = Integer.parseInt(num.substring(len, len+1));
				itw = dig * 2;
				newt = (itw % 10+ rem) + newt;
				len--;

				rem = itw / 10;
			}
			
			if(newt.charAt(0)=='0')
				num = newt.substring(1);																	
			else
				num = newt;
			System.out.println(i+"\t\t\t:\t\t\t\t"+num);
		}
		
		
		System.out.println(num);

		Tupac.main(null);
	}


}

File: tupac.py
'''
Created on 23-Aug-2014

@author: zubair-abid
'''
def binary():
    n = int(input("Enter index value: "))
    k = n*"0"
    return "1"+k


def integral():
    n = int(input("Enter index value: "))
    dig = 0; itw = 0 ; rem = 0 ; lent = 0;
    num = "1" ; newt = ""
    
    for i in range(1 , n + 1):
        newt = ""
        num = "0" + num
        lent = len(num) -1 ;
        rem = 0;    
        
        while lent >= 0:
            dig = num[lent : lent+1]
            itw = int(dig) * 2
            newt = str(itw % 10+ rem) + newt;
            lent -= 1
            
            rem = itw // 10;
        
        if newt[0]=='0' :
            num = newt[1: len(newt)+1];                                                                    
        else:
            num = newt;
        print(str(i)+"\t\t\t:\t\t\t"+num);
    return num


while True:
    integral()
        
